### ISSUE TITLE
No error/Info message when user tries to create a challenge (If he is removed from the project by someone)

### ISSUE THREAD

**Issue by codejamtc** [topcoder-platform/challenge-engine-ui#567]
https://monosnap.com/file/51507N0ozU4weyRQz0DctkGr2eMKZn

**URL:**
https://challenges.topcoder-dev.com/projects/16539/challenges/138f7b3b-6be1-43ee-b4ff-40856a798f07/edit

**Reproduction Steps:**

1. Open the application and Login as a valid user (jcori)
2. Select a Project (eg: Test & QA) > Click Launch New > Fill All > Click Create Challenge
3. Fill all the details in the form
4. Click "Launch as Draft"
5. Check the Draft challenges tab (It will display)
6. Now think some one removed (jcori) from project (Test & QA) - He desn't know

7. Select a Project (eg: Test & QA) > Click Launch New > Fill All > Click Create Challenge
8. Fill all the details in the form
9. Click "Launch as Draft"
10. Check the Draft challenges tab (Nothing displayed)

**Actual Result:** 

- No error/Info message when user tries to create a challenge (If he is removed from the project by someone)

**Expected Result:** 

- Should display an info message

**Device/OS:** 

- **Device(s):** Laptop
- **Resolution:** 1366 x 768
- **Operating System:** Windows 10 64bit (1909, 18363.815)

**Browser Information:** 

- **Browser(s):** Google Chrome 81.0.4044.138

**Comment by acshields** [comment]
@vikasrohit @mtwomey - What is your opinion on this issue? In my perspective, if a user is removed from the project in Connect they should A) not see the project in the Work Manager app and B) not be able to set-up/launch challenges against it.

Vikas, can you test this out & let me know if this still looks like an issue to you?

**Comment by vikasrohit** [comment]
Moving to December @acshields as we need to discuss this before starting on it.

**Comment by acshields** [comment]
@vikasrohit - noted. Let me know what I need to provide.

**Comment by vikasrohit** [comment]
@acshields it is tricky situation to deal with because this can happen with any application where some action has been performed out of that application and user is still haven't refreshed the original application and still looking at stale data. I don't know any standard solution for this and we need to discuss the situation in call.

**Comment by acshields** [comment]
@vikasrohit - noted. For now, I am going to move this to January. We can discuss later.

**Comment by acshields** [comment]
Updates here:

If a user has not refreshed their screen and has been removed from a project, when they click the "Launch New" button they should see this message:
- Primary Text: "You have been removed from the project and do not have permissions to launch new work."
- Sub-Text: "If you believe this is an error, please contact your project team members to have you added back to the project."
- 
- If a Copilot has been removed from a project and has a refreshed screen, their viewing permissions to these scenarios:

- They should not see the project as an option in the dropdown list for "Launch New"
- They can view active and completed challenges that they created for the project.
- They cannot view any challenges in New or Draft for the project.

If a Topcoder Employee has been removed from a project and has a refreshed screen, their viewing permissions to these scenarios:
- They should not see the project as an option in the dropdown list for "Launch New"
- They can still view all challenges under the "All Work," but the project and its challenges/tasks should no longer display under their "My Work"

**Comment by vikasrohit** [comment]
> Primary Text: "You have been removed from the project and do not have permissions to launch new work."

This is quite tricky because it is difficult to determine if the user previously had permission to view/launch challenges for the project and now not.

**Comment by yoution** [comment]
@phead198708 please assign to me

**Comment by yoution** [comment]
@vikasrohit
can we determine by `createby` field
![image](https://user-images.githubusercontent.com/3863914/124213856-e3602780-db23-11eb-9099-513c3285dce9.png)
if the chalenge is created by the user, but when BE return `no access`, when show the error?

**Comment by vikasrohit** [comment]
@yoution I didn't understand what we can determine with `createdBy` here. However, I think what we need is that we should try to reproduce the issue and in the step `Click "Launch as Draft"` what is the error returned by the API. If there is an error returned, we can act on that. If there is no error during the process, I guess we won't be fixing this issue.

**Comment by phead198708** [comment]
Hi, @vikasrohit  I suppose if one user send a create challenge request, the endpoints will check the permissions of this user first (for example, check he is in copilot list or not), if he has been removed from copilot list, the endpoint should return 403 access denied, right ? This will protect the challenge creation from the backend side.

**Comment by vikasrohit** [comment]
Yes, it should be something similar, however, I am not sure if that is happening right now. Can we verify what is exactly happening after user hits the Save as Draft button while user being removed from the project within same timeframe?

**Comment by yoution** [comment]
@vikasrohit , if the user not in the project, but can still `save draft`
<img width="1225" alt="View Details 2021-07-02 18-26-14" src="https://user-images.githubusercontent.com/3863914/124261233-21307080-db63-11eb-8401-34908b9a64ae.png">

**Comment by phead198708** [comment]
Hi, @yoution we need to add this limit to backend code. If user didn't have permission to create/edit the challenge, it should throw 403.

**Comment by vikasrohit** [comment]
> @vikasrohit , if the user not in the project, but can still `save draft`
> <img alt="View Details 2021-07-02 18-26-14" width="1225" src="https://user-images.githubusercontent.com/3863914/124261233-21307080-db63-11eb-8401-34908b9a64ae.png">

Ok, then why the challenge is not listed under `Drafts` tab as per the last step of issue reproduction?

>Hi, @yoution we need to add this limit to backend code. If user didn't have permission to create/edit the challenge, it should throw 403.

@phead198708 @yoution just to be clear, we are not attempting to fix anything in challenge api here. If we find that issue can not be fixed on the front end, we have to leave it. But we have to put our analysis results here why we think it can not be handled in front end.

**Comment by yoution** [comment]
@vikasrohit @phead198708  in the project list, the project doesn't exist, but when I input the url directly in the browser address, the user can edit the challenge in the project, can even active the challenge
![image](https://user-images.githubusercontent.com/3863914/124415945-82855900-dd88-11eb-9880-3e69f5e4c133.png)

**Comment by vikasrohit** [comment]
Ok, that is understandable. However, as per the original reproduction steps  the last step is `Check the Draft challenges tab (Nothing displayed)`, so if you don't go back to the project listing page after creating the challenge and instead land on the challenge listing page, do you see the challenge in Draft tab?

**Comment by yoution** [comment]
@vikasrohit yes, I can see it in the Draft tab, I can also active it from the Draft tab

**Comment by vikasrohit** [comment]
ok, Then I guess, we don't need any change for now. @codejamtc can you please verify that you are also seeing the same behavior?

**Comment by vikasrohit** [comment]
Moving to Backlog as it seems to be very edge case and we yet don't have exact requirements on how to handle it.
